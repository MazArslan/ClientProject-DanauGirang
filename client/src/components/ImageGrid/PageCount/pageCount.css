.counter {
	margin-top: 16px;
}
