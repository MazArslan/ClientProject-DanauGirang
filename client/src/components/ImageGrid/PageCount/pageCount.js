import React from "react";
import propTypes from "prop-types";
import "./pageCount.css";

/**
 * PageCount component.
 * Displays text informing user of their current page and the total number of pages
 * @param { currentPage, totalPages } props
 */
const pageCount = props => {
  const { currentPage, totalPages } = props;

  return (
    <React.Fragment>
      <div className="counter">
        <p>
          <i>
            Page {currentPage} out of {totalPages}
          </i>
        </p>
      </div>
    </React.Fragment>
  );
};

pageCount.propTypes = {
  currentPage: propTypes.string.isRequired,
  totalPages: propTypes.string.isRequired
};

export default pageCount;
