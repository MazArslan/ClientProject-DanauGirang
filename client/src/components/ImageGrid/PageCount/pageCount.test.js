import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import PageCount from "./pageCount";

Enzyme.configure({ adapter: new Adapter() });

describe("<PageCount/>", () => {
  const props = {
    currentPage: "3",
    totalPages: "5"
  };
  const container = shallow(<PageCount {...props} />);
  describe("component rendering", () => {
    it("should render correctly", () => {
      expect(container.exists()).toBe(true);
      expect(toJson(container)).toMatchSnapshot();
    });

    it("should render text", () => {
      const textBody = "Page 3 out of 5";
      const foundStringArray = container.find("i").props().children;
      expect(foundStringArray.join("")).toEqual(textBody);
    });
  });
});
