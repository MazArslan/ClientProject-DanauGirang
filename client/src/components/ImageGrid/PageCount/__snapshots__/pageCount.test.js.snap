// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageCount/> component rendering should render correctly 1`] = `
<Fragment>
  <div
    className="counter"
  >
    <p>
      <i>
        Page 
        3
         out of 
        5
      </i>
    </p>
  </div>
</Fragment>
`;
